doc/topics/blocking-vs-non-blocking.md

```diff
@@ -0,0 +1,140 @@
+# Overview of Blocking vs Non-Blocking
+
+This overview covers the difference between blocking and non-blocking calls in
+Node.js. This overview will refer to the event loop and libuv but no prior
+knowledge of those topics is required. Readers are assumed to have a basic
+understanding of the JavaScript language and Node.js callback pattern.
+
+> In this overview, "I/O" refers primarily to interaction with the system's disk
+and network supported by [libuv](http://libuv.org/).
+
+
+## Blocking
+
+Blocking is when the execution of additional JavaScript in the Node.js process
+must wait until an I/O operation completes. Blocking may occur when using any of
+the synchronous I/O methods in the Node.js standard library that use libuv. Use
+of blocking methods prevents the event loop from doing additional work while
+waiting for I/O to complete.
+
+In Node.js, JavaScript that exhibits poor performance due to being CPU intensive
+rather than I/O bound, isn't typically referred to as blocking.
+
+All of the I/O methods in the Node.js standard libraries provide asynchronous
+versions, which are non-blocking, and accept callback functions. Some methods
+also implement blocking methods, which usually have names that end with `Sync`.
```

Comment:
"Some methods also implement" - the methods don't implement other methods. Maybe "Some methods also have blocking counterparts implemented"?